.github/workflows/publish-master.yml

```
@@ -102,10 +102,97 @@ jobs:
+          curl https://raw.githubusercontent.com/helm/helm/main/scripts/get-helm-3 | bash
+
+      - name: Authenticate Helm to GHCR
+        run: |
+          echo "${{ secrets.GITHUB_TOKEN }}" | helm registry login ghcr.io --username $GITHUB_ACTOR --password-stdin
+
+
+      - name: Extract Helm Chart Version
+        id: extract-version
+        run: |
+          CHART_VERSION=$(yq eval '.version' ./helm/teraslice/Chart.yaml)
+          echo "CHART_VERSION=$CHART_VERSION" >> $GITHUB_ENV
+        env:
+          CHART_VERSION: ""
+
+      - name: Check if Helm Chart Version Exists in GHCR
+        id: check-version
+        env:
+          CHART_VERSION: ${{ env.CHART_VERSION }}
+        run: |
+          if curl -L -H "Accept: application/vnd.github+json" \
+          -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
+          -H "X-GitHub-Api-Version: 2022-11-28" \
+          "https://api.github.com/orgs/terascope/packages/container/teraslice-chart/versions" | \
+              jq -r --arg version "$CHART_VERSION" '.[] | select(.metadata.container.tags[] == $version)' | \
+              grep -q "$CHART_VERSION"; then
+              echo "Helm chart version $CHART_VERSION already exists in GHCR."
+              echo "version_exists=true" >> $GITHUB_ENV
+            else
+              echo "Helm chart version $CHART_VERSION does not exist in GHCR."
+              echo "version_exists=false" >> $GITHUB_ENV
+            fi
+
+      - name: Package and Update index.yaml
+        if: env.version_exists == 'false'
+        run: |
+          cd ./helm/helm-repo-site
+          echo $(helm package ../teraslice)
+          echo "Packaged Helm chart."
+          echo $(ls)
+          helm repo index . --merge=index.yaml --url oci://ghcr.io/terascope/teraslice-chart:${{ env.CHART_VERSION }}
+          echo $(cat index.yaml)
+          echo $(ls)
+          sed -i 's|/teraslice-${{ env.CHART_VERSION }}.tgz||' index.yaml
+          echo $(cat index.yaml)
+
+      - name: Commit Updated index.yaml
+        if: env.version_exists == 'false'
+        run: |
+          echo $(ls)
+          echo $(find ./helm/helm-repo-site/index.yaml)
+          echo $(cat ./helm/helm-repo-site/index.yaml)
+          git config user.name "github-actions[bot]"
+          git config user.email "github-actions[bot]@users.noreply.github.com"
+          git add ./helm/helm-repo-site/index.yaml
+          git commit -m "Update Helm chart index.yaml for version $CHART_VERSION"
+          git push origin master
+
+      - name: Publish Helm Chart to GHCR
+        if: env.version_exists == 'false'
+        run: |
+          cd ./helm/helm-repo-site
+          echo $(ls)
+          mv ./teraslice-${{ env.CHART_VERSION }}.tgz teraslice-chart-${{ env.CHART_VERSION }}.tgz
+          echo $(ls)
+          CHART_FILE=$(ls teraslice-chart-${{ env.CHART_VERSION }}.tgz)
+          echo "Publishing $CHART_FILE to GHCR..."
+          helm push $CHART_FILE oci://ghcr.io/terascope --debug
```

Should this line have the project name?

```suggestion
          helm push $CHART_FILE oci://ghcr.io/terascope/teraslice --debug
```

It seems unexpected that it's the organization.

---

Github container registries are only name-spaced by either an organization or a user account. For example the `base-image` docker image lives in the organization namespace along side the `teraslice` image. They're linked to their repos because of the workflows that publish them. 

https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry#pushing-container-images

> When you push a container image from the command line, the image is not linked to a repository by default. This is the case even if you tag the image with a namespace that matches the name of the repository, such as ghcr.io/octocat/my-repo:latest.
> 
> The easiest way to connect a repository to a container package is to publish the package from a workflow using ${{secrets.GITHUB_TOKEN}}, as the repository that contains the workflow is linked automatically. Note that the GITHUB_TOKEN will not have permission to push the package if you have previously pushed a package to the same namespace, but have not connected the package to the repository.
> 
> To connect a repository when publishing an image from the command line, and to ensure your GITHUB_TOKEN has appropriate permissions when using a GitHub Actions workflow, we recommend adding the label org.opencontainers.image.source to your Dockerfile. For more information, see “[Labelling container images](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry#labelling-container-images)” in this article and “[Publishing and installing a package with GitHub Actions](https://docs.github.com/en/packages/managing-github-packages-using-github-actions-workflows/publishing-and-installing-a-package-with-github-actions).” 

This can be verified using the gitHub api to request all packages that live in the namespace.
https://docs.github.com/en/rest/packages/packages?apiVersion=2022-11-28#list-packages-for-an-organization

Basically when ci pushes the tarball, it will set the repo to teraslice by default.